Block Supports: Add presets to dimension block supports

aaronrobertshaw opened the pull request:
> [!WARNING]
> This PR depends on https://github.com/WordPress/wordpress-develop/pull/10470 and https://github.com/WordPress/wordpress-develop/pull/10474 and should be rebased and merged after them.

Trac ticket: https://core.trac.wordpress.org/ticket/64413

This PR brings the changes from the following Gutenberg PRs to core:

https://github.com/WordPress/gutenberg/pull/73811

### Description

Adds support for preset dimensions values to be defined within theme.json and used across block supports such as; width, min-height etc.

### Testing

1. Adopt width support on a block or use the Group block with min-height support
2. Add `dimensionSizes` presets to your theme.json
3. Add styles to your test blocks within your theme.json using both the presets and custom values

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

aaronrobertshaw committed:
Block Supports: Add presets to dimension block supports

aaronrobertshaw committed:
Fix linting errors

aaronrobertshaw committed:
Add missing height to the dimensionSizes config

aaronrobertshaw committed:
Add missing css_vars for height in style engine config

andrewserong commented on src/wp-includes/style-engine/class-wp-style-engine.php:
```diff
@@ -210,14 +210,38 @@ final class WP_Style_Engine {
+				'property_keys' => array(
+					'default' => 'width',
 				),
+				'path'          => array( 'dimensions', 'width' ),
```
This might just be a bad rebase, but the `width` key has been duplicated here and the latter wins out. When I remove the latter, the width preset is output correctly.

The good news is that the unit test you added is failing as expected for this one 😄

ramonjd replied:
I'll remove the last one in the upcoming rebase. Hope I don't break anything

aaronrobertshaw replied:
Bad rebased, sorry. Thanks for fixing that up.

andrewserong commented on tests/phpunit/tests/style-engine/styleEngine.php:
```diff
@@ -119,6 +119,21 @@ public function data_wp_style_engine_get_styles() {
+			'inline_valid_dimension_preset_style'          => array(
+				'block_styles'    => array(
+					'dimensions' => array(
+						'width' => 'var:preset|dimension|large',
```
Tiny nit: would it be worth including the `height` key here, too?

ramonjd replied:
I'll add it

andrewserong commented on tests/phpunit/tests/theme/wpThemeJson.php:
```diff
@@ -510,6 +528,7 @@ public function test_get_stylesheet() {
 							),
 							'dimensions' => array(
 								'minHeight' => '50vh',
+								'width'     => 'var:preset|dimension|large',
```
For completeness sake, would it be worth adding height here, too, perhaps with a static value like `500px`?

ramonjd replied:
I'll do this too

andrewserong reviewed (commented):
Looking really good — looks like there's just one issue to fix up (I've left a comment), but when I updated the `width` object locally, everything tested well. Once that's sorted I think this will be good to land (and the preceding two PRs that this builds on).

Left a couple of tiny nits re: the tests, just if we should ensure we add both width and height in the test cases, rather than just using one of them. Not a blocker at all, though.

ramonjd committed:
Remove unused width configuration from style engine dimensions

ramonjd committed:
Update style engine and theme JSON tests to include height dimension presets

- Added height dimension preset to the style engine test cases.
- Updated expected CSS output to reflect the inclusion of height in the style engine configuration.
- Modified theme JSON tests to incorporate height dimension preset for consistency in styling.

ramonjd committed:
- Changed height from preset value to a fixed pixel value of 500px in theme JSON tests.

ramonjd commented:
I ran out of time to test this properly, sorry, but it's rebased and I addressed the feedback.

andrewserong reviewed (approved):
This is re-testing nicely for me:

Site frontend:

<img width="1169" height="597" alt="Image" src="https://github.com/user-attachments/assets/f6a4344f-e346-41c2-bb7c-e942439889e3" />

And in the editor:

<img width="1276" height="649" alt="Image" src="https://github.com/user-attachments/assets/78e824ab-61ca-46da-aade-57a88ac45bed" />

Just ignore my design choices 😄 
LGTM! 🚀

ramonjd commented:
Committed in [r61631](https://core.trac.wordpress.org/changeset/61631) and https://github.com/WordPress/wordpress-develop/commit/d0a95382fe57a7f7452e9e031dd713e285898464
